test/spec/features/outline/MultiSelectionOutlineSpec.js

```diff
@@ -0,0 +1,168 @@
+import {
+  bootstrapDiagram,
+  inject,
+} from 'test/TestHelper';
+
+import outlineModule from 'lib/features/outline';
+import modelingModule from 'lib/features/modeling';
+
+import {
+  classes as domClasses,
+  query as domQuery
+} from 'min-dom';
+
+import {
+  resizeBounds
+} from 'lib/features/resize/ResizeUtil';
+
+import { getBBox } from 'lib/util/Elements';
+
+
+describe('features/outline - MultiSelectionOutline', function() {
+
+  beforeEach(bootstrapDiagram({ modules: [ outlineModule, modelingModule ] }));
+
+
+  var shape1, shape2, selectedShapes, outline, bounds;
+
+  beforeEach(inject(function(elementFactory, canvas, selection) {
+    shape1 = elementFactory.createShape({
+      id: 'shape1',
+      x: 100, y: 100, width: 100, height: 100
+    });
+
+    shape2 = elementFactory.createShape({
+      id: 'shape2',
+      x: 300, y: 100, width: 100, height: 100
+    });
+
+    canvas.addShape(shape1);
+    canvas.addShape(shape2);
+
+    selectedShapes = [ shape1, shape2 ];
+
+    selection.select(selectedShapes);
+
+    outline = domQuery('.djs-selection-outline');
+
+    bounds = getBounds(outline);
+  }));
+
+
+  it('should show box', inject(function() {
+    expect(outline).to.exist;
+  }));
+
+
+  it('should add djs-multi-select marker', inject(function(selection, canvas) {
+
+    // when
+    var element = canvas.getContainer();
+
+    // then
+    expect(domClasses(element).has('djs-multi-select')).to.be.true;
+
+    // but when
+    selection.select(null);
+
+    // then
+    expect(domClasses(element).has('djs-multi-select')).to.be.false;
+  }));
+
+
+  it('should not add djs-multi-select marker for single selection', inject(function(canvas) {
+
+    // when
+    var element = canvas.getContainer();
+
+    // then
+    expect(domClasses(element).has('djs-multi-select'));
```

Comment:
```suggestion
    expect(domClasses(element).has('djs-multi-select')).to.be.false;
```

Reply:
General note: 

* I'd rather not have vast test preparation in a `beforeEach` block. This particular test for example is not understandable to me, I need to loop back to `beforeEach` to understand the `// given` aspects, which include [pre-selection and bounds processing](https://github.com/bpmn-io/diagram-js/pull/943/files#diff-7cc26e6f775cc528290b2959bd56a6ca071cbb8ef47898bb8f6e1f43a7b45c8eR42-R48). Better: Everything that is related to the behavior tested should be inside the test. You can of course wrap re-usable things in helper functions, if you desire, i.e.:

  ```
  const { bounds } = select(elements)
  ```

Reply:
I'll push an update to visualize this in a minute.

Reply:
See https://github.com/bpmn-io/diagram-js/pull/943/commits/01924e56520fe4195a3a6bbd19c346b347b258d6 and https://github.com/bpmn-io/diagram-js/pull/943/commits/03b026f4f439c3d0c7fd7f571a15ebb3475d5729.

Tests should be as isolated as possible, and extensive essential setup in `beforeEach` is an anti-pattern we want to avoid. Usually it is trivial to wrap setup and/or assertions in a helper so that they can be re-used across specs, as seen [here](https://github.com/bpmn-io/diagram-js/pull/943/commits/03b026f4f439c3d0c7fd7f571a15ebb3475d5729#diff-7cc26e6f775cc528290b2959bd56a6ca071cbb8ef47898bb8f6e1f43a7b45c8eR104) for example.

Reply:
I think it can be all squashed together. The changes are concise and are not that huge, we have 11 commits just because of me learning